src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -730,16 +733,16 @@ - (void)writeAttributeWithEndpointID:(NSNumber *)endpointID
                               completion:^(NSArray<NSDictionary<NSString *, id> *> * _Nullable values, NSError * _Nullable error) {
                                   MTR_LOG_INFO("%@ completion error %@ endWork", logPrefix, error);
                                   [workItem endWork];
+                                  if (error) {
+                                      [self removeExpectedValuesForAttributePath:attributePath];
```

Comment:
There's an edge case here where we write value A for a path, then write value B, then the first write fails and the second write succeeds.  Is it worth tracking some sort of "we are responsible for this expected value" state, or should we just not worry about this case?

Reply:
The latest change addresses this by associating expected values with an ID, return it, and have removal confirm the ID to remove.